rust/crd/src/authentication.rs

```diff
@@ -101,8 +114,22 @@ impl ResolvedAuthenticationClasses {
 
         for auth_class in &self.resolved_authentication_classes {
             match &auth_class.spec.provider {
+                // explicitly list each branch so new elements do not get overlooked
                 AuthenticationClassProvider::Tls(_) => {}
-                _ => {
+                AuthenticationClassProvider::Kerberos(_) => {}
+                AuthenticationClassProvider::Static(_) => {
+                    return Err(Error::AuthenticationProviderNotSupported {
+                        authentication_class: ObjectRef::from_obj(auth_class),
+                        provider: auth_class.spec.provider.to_string(),
+                    })
+                }
+                AuthenticationClassProvider::Ldap(_) => {
+                    return Err(Error::AuthenticationProviderNotSupported {
+                        authentication_class: ObjectRef::from_obj(auth_class),
+                        provider: auth_class.spec.provider.to_string(),
+                    })
+                }
+                AuthenticationClassProvider::Oidc(_) => {
```

Comment:
You can combine the blocks as they all contain the same content
```suggestion
                AuthenticationClassProvider::Static(_)
                | AuthenticationClassProvider::Ldap(_)
                | AuthenticationClassProvider::Oidc(_) => {
```